src/Forms/FormAction.php

```diff
@@ -93,9 +97,45 @@ public function getSchemaDataDefaults()
         $defaults = parent::getSchemaDataDefaults();
         $defaults['attributes']['type'] = $this->getUseButtonTag() ? 'button' : 'submit';
         $defaults['data']['icon'] = $this->getIcon();
+        $defaults['data']['savedTitle'] = $this->getSavedTitle();
+        $defaults['data']['savedClasses'] = $this->getSavedClasses();
+        $defaults['data']['savedIcon'] = $this->getSavedIcon();
         return $defaults;
     }
 
+    public function setSavedTitle($title)
```

Comment:
None of these methods describe what each of these are used for; They also seem to be very specific use cases baked into the generic form action class, which feels a little over-zealous.

You could generalise it, so you don't need so many extra methods into a saved state?

```php
FormAction::create('save', _t(__CLASS__.'.SAVE', 'Save'))
    ->setIcon('save')
    ->setSavedState([
    	'title' => _t(__CLASS__.'.SAVED', 'Saved'),
    	'icon' => 'tick',
    	'extraClass' => 'primary-outline',
   	]);
```
and in js
```js
getProps() {
	let props = Object.assign({}, this.props, this.props.data);
	if (this.changed) {
		props = Object.assign({}, this.props.data.savedState);
	}
	return props;
}
```

Or something :P